distributed/scheduler.py

```diff
@@ -620,6 +624,18 @@ def __init__(self, key, run_spec):
         self.loose_restrictions = False
         self.actor = None
         self.type = None
+        self.group_key = key_split_group(key)
+        self.group = None
+
+    @property
+    def state(self):
+        return self._state
+
+    @state.setter
+    def state(self, value):
+        self.group.states[self._state] -= 1
+        self.group.states[value] += 1
```

Comment:
It would, yes.  Does this pose a significant problem for you?

Reply:
I was mostly thinking about the overall lifecycle of these objects (TG/TP) and noticed that they will never be removed but these dicts are properly bound so I don't think this is an issue. This might be an unwanted data artefact but I guess the forgotten state is more or less irrelevant

Reply:
Would make a difference when exposing this directly as a prometheus gauge but this can be fixed over there as well , e.g. by only exporting `ALL_TASK_STATES`

Reply:
This will also give you a measure how of many tasks you've processed over all time, which may be of interest.  You can also look at the `active_states` attribute of the TaskPrefix to exclude all groups that have been fully forgotten.

I'm inclined to try it this way first, and then if we find that the forgotten task groups accumulate in an inconvenient way then we remove them.